go/service/main.go

```diff
@@ -470,6 +471,31 @@ func (d *Service) runBackgroundIdentifierWithUID(u keybase1.UID) {
 	d.G().AddUserChangedHandler(newBgi)
 }
 
+func (d *Service) runBackgroundPerUserKeyUpgrade() {
+	if !d.G().Env.GetUpgradePerUserKey() {
+		d.G().Log.Debug("PerUserKeyBackground disabled (not starting)")
+		return
+	}
+
+	arg := &engine.PerUserKeyBackgroundArgs{
+		Settings: engine.PerUserKeyBackgroundDefaultSettings,
+	}
+	eng := engine.NewPerUserKeyBackground(d.G(), arg)
+	go func() {
+		ectx := &engine.Context{NetContext: context.Background()}
+		err := engine.RunEngine(eng, ectx)
+		if err != nil {
+			d.G().Log.Warning("per-user-key background upgrade error: %v", err)
+		}
+	}()
```

Comment:
Yes good point, it is redundant. Also this led me to find that the inner `PerUserKeyUpgrade` engine is being run with the wrong context and wouldn't listen to `cancel`.